lib/lightstep/propagation.rb

```diff
@@ -6,14 +6,15 @@
 module LightStep
   module Propagation
     PROPAGATOR_MAP = {
-      lightstep: LightStepPropagator
+      lightstep: LightStepPropagator,
+      b3: B3Propagator
     }
 
     class << self
       # Constructs a propagator instance from the given propagator name. If the
       # name is unknown returns the LightStepPropagator as a default
       #
-      # @param [Symbol, String] propagator_name
+      # @param [Symbol, String] propagator_name One of :lightstep or :b3
```

Comment:
Would it make sense to instead take the propagator itself, so the argument is stringly-typed (/symbolly-typed 😂)?

Reply:
I had it setup like that originally. I changed it for two reasons. Configuring the propagator is slightly more cumbersome. You have to do this:

```
LightStep.configure(component_name: 'lightstep/ruby/example', access_token: 'your_access_token', propagator: LightStep::Propagation::B3Propagator.new)
```

Instead of this:

```
LightStep.configure(component_ name: 'lightstep/ruby/example', access_token: 'your_access_token', propagator: :b3)
```

The other and more important reason I changed it is that we ultimately want to allow configuration via an environment variable. Being able to specify the propagator as a string or symbol gets you half the way there.

If you would like to be able to specify the propagator directly, I'd recommend that we accept a string, symbol, or propagator, but I don't know that complexity is worth it.